### PR TITLE
Add handbook documentation for testing, local config, and style issues

### PR DESCRIPTION
**Pull request by brianjgeiger**
## Purpose

Make the handbook better

## Summary of Changes

1. Add notes on testing (expanded from the notion page). Pay extra attention here as I may have lied once or twice
2. Add dev-env section back in with instructions on setting up local overrides to configuration
3. Steal the visual style info from osf-style
4. Steal the written style info from osf-style

## QA Notes

No QA necessary unless you want to complain about some of the things that I stole from osf-style.

## Ticket

N/A

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] ~~testable and includes test(s)~~ No tests
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/handbook/addon/docs/testing/template.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
 # Testing
 
-TODO
+## Test categories
+
+### By what they try to determine
+We organize the tests in the hierarchy by this set of categories.
+
+#### Acceptance test
+These are a type of Application test (see below) that looks at the behavior of the whole app from a user's perspective. Ideally, when adding a new page, the Acceptance tests should be the first thing you write after you get all of the appropraite information from Product and preferably after consulting with QA. The handy thing about acceptance tests is that it's behavior driven, so it shouldn't matter what the underlying implementation of the models or the components are. If the OSF API is defined, you should be able to define what the test selectors are and how they react to various situations, then fill in the functionality of everything else to make those tests pass.
```
Comment (aaxelb) [inline lib/handbook/addon/docs/testing/template.md]: s/appropraite/appropriate

Comment (aaxelb) [inline lib/handbook/addon/docs/testing/template.md]: > If the OSF API is defined, you should be able to define what the test selectors are and how they react to various situations, then fill in the functionality of everything else to make those tests pass.

I think it would be clearer to move this to a new paragraph and expand on it slightly. If we hadn't already talked about it, I might have trouble understanding what you mean.

File: lib/handbook/addon/docs/testing/template.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+
+If your Acceptance test isn't robust enough to survive a refactor of the ember-osf-web application (to typescript, to a new version of ember, to engines, or whatever), then it's probably not been designed right. Good tests here will eliminate a lot of testing that QA would have to otherwise do.
+
+You will make heavy use of [Mirage](http://www.ember-cli-mirage.com) for these tests.
+
+#### Unit test
+They are tests that look at behavior of a single thing and are mostly used for services, utils, serializers, and controller actions. They are very useful for verifying interfaces of logic that need to be consistent regardless of how it will ultimately be used. Consequently, they are super handy if you know that your code will one day need to act a certain way, but nothing uses that functionality just yet, that you can be assured that it will be ready when the day arrives. Unit tests should be reasonably quick to create and fast to run, so completeness is ofen desirable in a unit test. Try to get all of the edge cases here rather than, say, in the Acceptance tests.
```
Comment (aaxelb) [inline lib/handbook/addon/docs/testing/template.md]: s/ofen/often

File: lib/handbook/addon/docs/testing/template.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
-TODO
+## Test categories
+
+### By what they try to determine
+We organize the tests in the hierarchy by this set of categories.
+
+#### Acceptance test
+These are a type of Application test (see below) that looks at the behavior of the whole app from a user's perspective. Ideally, when adding a new page, the Acceptance tests should be the first thing you write after you get all of the appropraite information from Product and preferably after consulting with QA. The handy thing about acceptance tests is that it's behavior driven, so it shouldn't matter what the underlying implementation of the models or the components are. If the OSF API is defined, you should be able to define what the test selectors are and how they react to various situations, then fill in the functionality of everything else to make those tests pass.
+
+If your Acceptance test isn't robust enough to survive a refactor of the ember-osf-web application (to typescript, to a new version of ember, to engines, or whatever), then it's probably not been designed right. Good tests here will eliminate a lot of testing that QA would have to otherwise do.
```
Comment (aaxelb) [inline lib/handbook/addon/docs/testing/template.md]: > Good tests here will eliminate a lot of testing that QA would have to otherwise do.

Or, more accurately, will prevent additional rounds of development and QA testing. (Sounds like you meant acceptance tests kinda replace QA, which I don't think is true)

File: lib/handbook/addon/docs/visual-style/template.md [hunk 4]
```diff
@@ -1,5 +1,98 @@
...
+- Most of the time select and checkbox items are too small, consider increasing their size or generally make sure they fill your font's line height.
+- Make sure tab goes to the next item if there is a form with multiple items.
+- If using JS for form inputs, bind Enter key.
+- Don't have a "Clear Form" button.
+- Avoid using too many borders in form structures. Alignment is a way to denote edges as well and inputs already have borders.
+- Use a good combination of placeholder and label for inputs, often one is enough, both are unnecessary. Don't repeat the label in the placeholder. Label is better for cross browser use.
+- Tie your labels to your form elements with "for" so that when users click on the label the input gets selection.
+
+### Respect history
+If you are using a single page approach where page loads are not required for significant events, remember that users consider significant layout changes to denote page changes and will want to press back to go back to the previous state.
```
Comment (aaxelb) [inline lib/handbook/addon/docs/visual-style/template.md]: Doesn't make sense to specify this in an ember app, it should happen automatically.

File: lib/handbook/addon/docs/visual-style/template.md [hunk 5]
```diff
@@ -1,5 +1,98 @@
 # Visual Style
 
-TODO
+## General principles
```
Comment (aaxelb) [inline lib/handbook/addon/docs/visual-style/template.md]: I don't know how I feel about copying these over as-is... but I guess it's better to have them here and maybe revise them over time, if it makes sense to.

Comment (jamescdavis) [reply lib/handbook/addon/docs/visual-style/template.md]: It's a place to start?...

File: lib/handbook/addon/docs/resources/template.md [hunk 6]
```diff
@@ -12,3 +12,17 @@
...
+
+## Various Addons
+- [ember-a11y-testing](https://github.com/ember-a11y/ember-a11y-testing) - Catching easy-to-miss, easy-to-detect accessibility problems
+- [ember-cli-addon-docs](https://github.com/ember-learn/ember-cli-addon-docs) - That's this handbook
+- [ember-cli-mirage](http://www.ember-cli-mirage.com) - Fake API responses for easy testing and development
+- [ember-concurrency](http://ember-concurrency.com/docs/introduction/) - Tasks for asynchronous actions
+- [ember-css-modules](https://github.com/salsify/ember-css-modules) - Don't style globally when you can style locally
+- [ember-decorators](https://github.com/ember-decorators/ember-decorators) - Save a lot of trouble coding… with decorators!
+- [ember-engines](https://github.com/ember-engines/ember-engines) - How we separate code now
+- [ember-feature-flags](https://github.com/kategengler/ember-feature-flags) - For waffle feature flags
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply lib/handbook/addon/docs/resources/template.md]: I mostly added that because we often talk about "waffle flags" instead of "feature flags`" and just saying "For feature flags" would be super redundant with the name.

File: lib/handbook/addon/docs/testing/template.md (hunk 1 above)
Comment (brianjgeiger) [reply lib/handbook/addon/docs/testing/template.md]: So _apparently_ VSCode doesn't spell check by default. Interesting.

File: lib/handbook/addon/docs/testing/template.md (hunk 3 above)
Comment (brianjgeiger) [reply lib/handbook/addon/docs/testing/template.md]: No, I think it will eliminate the need for some testing work if it's done well and is reliable. That testing work may just be creation of selenium tests, but it might also be a lot of regression testing.

File: lib/handbook/addon/docs/visual-style/template.md (hunk 5 above)
Comment (brianjgeiger) [reply lib/handbook/addon/docs/visual-style/template.md]: Yeah, we should definitely think about ways to make these better, but since we want to be away from osf-style in any case, at least having it in our repo means we can easily make these changes.

**Comment by brianjgeiger** [comment]
@aaxelb @jamescdavis Okay, this is updated with most of the suggested changes. Time to find a spell checking extension to vscode.

File: lib/handbook/addon/docs/resources/template.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jamescdavis) [reply lib/handbook/addon/docs/resources/template.md]: The Ember app doesn't refer to them as waffle flags anywhere and even the API just calls them `active_flags` (at least in the endpoint we use). IMHO we should stop calling them waffle flags because waffle is a back-end implementation detail.

**Comment by jamescdavis** [comment]
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Comment by brianjgeiger** [comment]
Installed

File: lib/handbook/addon/docs/resources/template.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jamescdavis) [reply lib/handbook/addon/docs/resources/template.md]: Also, I'd like to start using `ember-feature-flags` for more than just flags we get from the API. For example, I'd like to start using it for https://www.branchbyabstraction.com
